packages/component-submission/server/use-cases/helpers/files.js

```diff
@@ -129,59 +163,34 @@ class FilesHelper {
       })
     }
 
-    const semanticExtractionEntity = SemanticExtractionModel.createTitleEntity(
-      manuscriptId,
-      title,
-    )
-    await semanticExtractionEntity.save()
-
     return title
   }
 
-  static async cleanOldManuscript(manuscript) {
-    const oldFileIndex = manuscript.files.findIndex(
-      element => element.type === 'MANUSCRIPT_SOURCE',
-    )
-
-    if (oldFileIndex >= 0) {
-      logger.info(
-        `Manuscript Upload found index ${oldFileIndex} ${
-          manuscript.files[oldFileIndex].filename
-        } | ${manuscript.id}`,
-      )
-      const oldFile = await FileModel.find(manuscript.files[oldFileIndex].id)
-      manuscript.files.splice(oldFileIndex, 1)
-      await oldFile.delete()
-    }
-  }
-
-  static async setManuscriptMetadata(initialManuscript, title) {
-    return new Promise(async resolve => {
-      const manuscript = initialManuscript
-      const pendingFileIndex = manuscript.files.findIndex(
-        element => element.type === 'MANUSCRIPT_SOURCE_PENDING',
-      )
-      manuscript.files[pendingFileIndex].type = 'MANUSCRIPT_SOURCE'
-      logger.info(
-        `Manuscript Upload Manuscript::save ${title} | ${manuscript.id}`,
-      )
-      manuscript.meta.title = title
-      await manuscript.save()
-      resolve(manuscript)
-    })
+  static async swapPendingToSource(fileList) {
+    await cleanOldManuscript(fileList)
+    return setPendingToSource(fileList)
   }
 
-  static validateManuscriptSource(manuscript) {
-    const sourceList = manuscript.files.filter(
-      f => f.type === 'MANUSCRIPT_SOURCE',
-    )
-    const pendingList = manuscript.files.filter(
+  static validateManuscriptSource(fileList) {
```

Comment:
Function now returns a boolean. This is a FileHelper function and as such has (or should have) no idea of the related Manuscript id ... its the responsibility of the caller to  throw an error or handle the validation accordingly.